src/lib/consensus/proof_of_stake.ml

```diff
@@ -1856,9 +1856,14 @@ module Data = struct
             (Global_slot.create ~epoch:next_epoch ~slot:next_slot)
       ; checkpoints }
 
-    module M = Snarky.Snark.Run.Make (Curve_choice.Tick_backend)
+    module M =
+      Snarky.Snark.Run.Make
+        (Curve_choice.Tick_backend)
+        (struct
+          type t = unit
+        end)
 
-    let m : M.field Snarky.Snark.m = (module M)
+    let m : (unit, M.field) Snarky.Snark.m = (module M)
```

Comment:
Maybe we could have a version of `Snarky.m` that just has the prover state type abstract ? It would be nice not to have to talk about that type when we don't care what it is.

Reply:
@imeckler good idea, I've implemented this in o1-labs/snarky#271